pinc/misc.inc

```diff
@@ -1347,7 +1362,7 @@ function get_upload_err_msg($upload_error_code)
 /**
  * Represent bytes in the largest reasonable units
  */
-function humanize_bytes($n)
+function humanize_bytes(float $n): string
```

Comment:
Is `float` a superset of `int`? Because in project.php we call it with an `int`:
```php
function echo_byte_size(int $size): void
{
    echo " (", humanize_bytes($size), ")";
}
```

Reply:
`float` is not a superset of `int` from the type system's perspective. However `int` can be promoted to a `float` without loss.

The change is done because we do a float division on l. 1371 that involves `$n` (and we return potentially 2 decimal point) so it makes sense to use the wider type in the signature.